src/main/tv/codely/scala_intro_examples/lesson_08_collections/WordAggregator.scala

```diff
@@ -4,6 +4,10 @@ final class WordAggregator {
   // @ToDo: Here you have your exercise!
   // @see tv.codely.scala_intro_examples.lesson_08_collections.WordAggregatorSpec
   def aggregateWords(text: String): Map[String, Int] = {
-    Map.empty
+    if (text.length == 0) Map.empty[String,Int]
+    else
+      text.split("\\W+").foldLeft(Map.empty[String, Int]) {
+        (res, word) => res + (word.toLowerCase -> (res.getOrElse(word.toLowerCase, 0) +1 ))
```

Comment:
Nice approach!

Just a little suggestion: It would be nice to bring up some semantics around the `res` value. Something like `aggregatedWords` would do the work in order to denote that there we're storing the already aggregated words and using it as an accumulator. What do you think?